packages/fetchai/skills/erc1155_skill/handlers.py

```diff
@@ -48,17 +48,19 @@ def handle(self, message: Message) -> None:
         if tx_msg_response.tx_id == "contract_deploy":
             tx_signed = tx_msg_response.signed_payload.get("tx_signed")
             ledger_api = self.context.ledger_apis.apis.get("ethereum")
-            result = ledger_api.send_raw_transaction(  # type: ignore
-                tx_signed
+            result = ledger_api.send_signed_transaction(  # type: ignore
+                True, tx_signed
             )
             self.context.logger.info(result)
-            while contract.instance.address is None:
-                contract.update_contract_instance(
-                    contract_address=result.contractAddress,
-                    ledger_api=self.context.ledger_apis.apis.get("ethereum"),
-                )
-                time.sleep(3)
-            contract.is_deployed = True
+            contract.set_address(ledger_api, result)
```

Comment:
this should not be fed the `result` var, it should be fed the `address`

Reply:
At this point you don't have the address. You are returning the tx_digest from send_signed_transaction and not the attributeDict or the contract address.